zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/DeploymentReconstructionStarter.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.processing.deployment;
+
+import io.camunda.zeebe.engine.state.immutable.DeploymentState;
+import io.camunda.zeebe.protocol.impl.record.value.deployment.DeploymentRecord;
+import io.camunda.zeebe.protocol.record.intent.DeploymentIntent;
+import io.camunda.zeebe.stream.api.ReadonlyStreamProcessorContext;
+import io.camunda.zeebe.stream.api.StreamProcessorLifecycleAware;
+import io.camunda.zeebe.stream.api.scheduling.Task;
+import io.camunda.zeebe.stream.api.scheduling.TaskResult;
+import io.camunda.zeebe.stream.api.scheduling.TaskResultBuilder;
+import java.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class DeploymentReconstructionStarter implements StreamProcessorLifecycleAware, Task {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DeploymentReconstructionStarter.class);
+  private final DeploymentState deploymentState;
+
+  public DeploymentReconstructionStarter(final DeploymentState deploymentState) {
+    this.deploymentState = deploymentState;
+  }
+
+  @Override
+  public void onRecovered(final ReadonlyStreamProcessorContext context) {
+    if (deploymentState.hasStoredAllDeployments()) {
```

Comment:
❌ I don't think this check is safe? What happens if it already started the reconstruction and the broker restarts. The flag isn't set yet, as it'll get set once evrything is reconstructed. I think we need a flag to indicate if the reconstruction is started 🤔

Reply:
True, I did not think of this :+1:

Reply:
We might still write the initial Reconstruct command, then restart before processing it and write it again. We could mark the first one as such and then reject the second one if we've already started, _but:_ we could also just ignore this? At worst, we'd write a redundant Reconstruct command. But that wouldn't cause any conflicts or inconsistencies because the Reconstruct command doesn't carry any state and ends up as a no-op if there's nothing to do?

Reply:
As discussed, this is difficult to avoid and there's no harm in writing a redundant Reconstruct command either. Let's leave it for now.